app/validators/routes_and_redirects_validator.rb

```diff
@@ -147,11 +147,37 @@ def validate(location, redirect)
 
   private
 
+    def acceptable_destination?(target)
+      target.starts_with?("/") || is_valid_govuk_campaign_url?(target)
+    end
+
+    def is_valid_govuk_campaign_url?(target)
+      uri = URI.parse(target)
+      host = uri.host
+      if (host =~ /\A.+\.campaign\.service\.gov\.uk\z/i) != nil &&
+          %w(http https).include?(uri.scheme)
+        label = host&.split(".")&.first
```

Comment:
I am not sure in what scenario you wouldn't have a host at this point? Seems like the safe navigation is redundant? (and also possible a code smell :))